api/src/main/java/org/openmrs/module/appointments/model/AppointmentRecurringPattern.java

```diff
@@ -82,4 +82,23 @@ public Boolean isAppointmentsEmptyOrNull (){
                 .filter(appointment -> !appointment.getVoided()).collect(Collectors.toSet());
     }
 
+    public Set<Appointment> getRelatedAppointments() {
+        Set<Appointment> relatedAppointments = new LinkedHashSet<>();
+        getActiveAppointments().stream().forEach(appointment -> {
+            if (appointment.getRelatedAppointment() != null) {
+                relatedAppointments.add(appointment.getRelatedAppointment());
+            }
+        });
+        return relatedAppointments;
+    }
+
+    public Set<Appointment> getRemovedAppointments() {
+        Set<Appointment> removedAppointments = new LinkedHashSet<>();
+        Set<Appointment> relatedAppointments = getRelatedAppointments();
+        getAppointments().stream().forEach(appointment -> {
+            if (appointment.getVoided() == Boolean.TRUE && !relatedAppointments.contains(appointment))
```

Comment:
`appointment.getVoided()` enough?

Reply:
No. Because related appointments are also voided we don't want them.

Reply:
I mean boolean check `appointment.getVoided() == Boolean.TRUE`